api/rest/v1/middleware_deprecation_test.go

```diff
@@ -0,0 +1,20 @@
+package rest
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeprecationWrapper(t *testing.T) {
+	s := Server{}
+
+	handler := s.middlewareDeprecation(s.handleHello())
+
+	request, _ := http.NewRequest("GET", "/", nil)
+	response := httptest.NewRecorder()
+	handler.ServeHTTP(response, request)
+	assert.Equal(t, "true", response.Header().Get("Deprecated"), "True is expected")
```

Comment:
I would change this description to something more descriptive. 

```suggestion
	assert.Equal(t, "true", response.Header().Get("Deprecated"), "'Response contains deprecated header")
```